build_tools/teatime.py

```diff
@@ -81,6 +81,12 @@ def __init__(self, args: argparse.Namespace):
             self.log_file = open(self.log_path, "wb")
         self.log_timestamps: bool = args.log_timestamps
 
+        # S3 upload configuration
+        self.upload_to_s3 = os.getenv("TEATIME_S3_UPLOAD", "0") == "1"
+        self.s3_bucket = os.getenv("TEATIME_S3_BUCKET")
+        self.s3_subdir = os.getenv("TEATIME_S3_SUBDIR")
+        self.base_build_dir = os.getenv("BASE_BUILD_DIR")
```

Comment:
Might as well check that if one of `s3_bucket` or `s3_subdir` is set, the other must also be set, rather than silently fail below in that case:

```python
        # Call coordinate_index_and_logs if upload is configured
        if self.upload_to_s3 and self.s3_bucket and self.s3_subdir:
            self.coordinate_index_and_logs()
```

Actually, looking at this more closely, we shouldn't need `base_build_dir` here, but we _do_ need `log_path`. The `--build-dir` argument in `create_log_index.py` should be changed to `--logs-dir`. Rather than pass around the _build_ directory. That should result in some good simplification across the scripts and yaml files.